src/App.tsx

```diff
@@ -15,48 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  default = 0,
+  alphabet = 1,
+  length = 2,
+}
+
+interface FilterParams {
+  sortField: SortType;
+  isReverseField: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReverseField }: FilterParams,
+) {
+  let preparedgoods = [...goods];
+
+  if (sortField > 0) {
+    preparedgoods.sort((good1, good2) => {
+      switch (sortField) {
+        case 1:
+          return good1.localeCompare(good2);
+
+        case 2:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverseField) {
+    preparedgoods = preparedgoods.reverse();
+  }
+
+  return preparedgoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(0);
+  const [isReverseField, setIsReverseField] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, isReverseField });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(1)}
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortField !== 1 },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(2)}
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'is-light': sortField !== 2 },
+          )}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => (
+            setIsReverseField(!isReverseField)
+          )}
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button is-warning',
+            { 'is-light': !isReverseField },
+          )}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(!!sortField || isReverseField)
+        && (
+          <button
+            onClick={() => {
+              setSortField(0);
+              setIsReverseField(false);
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here